cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/StructuredTable.java

```diff
@@ -89,7 +89,7 @@
    * To match a non-existent value, the value of the expected field should be null.
    *
    * @param keys the primary key of row
-   * @param oldValue the expected value in the table
+   * @param oldValue the expected value in the table. The field cannot be part of the primary key
```

Comment:
I think you also need to modify the code to reject a compareAndSwap() for a primary key field, and same for increment().